### DISCUSSION TITLE
Warning when returning disposed object

### DISCUSSION THREAD

**Discussion by matthiaslischka** [dotnet/runtime#64283] [Ideas]
Returning a disposed object never makes sense, right? 
Or are there any scenarios where it does?

Something like that `using` keyword here is very tricky to spot and dangerous.

```csharp
public Stream GetFile()
{
 	using var fileStream = File.Create(@"c:\tmp\file.txt");
	return fileStream;
}
```

I was just wondering why there is no compiler warning or maybe I just missed it or have to configure it somehow?

BR Matthias
p.s.: [original request from dotnet/roslyn](https://github.com/dotnet/roslyn/discussions/58998)

**Comment by huoyaoyuan** [comment]
The compiler shouldn't warn about this because it doesn't care about object lifetime on heap.
Currently this rule conflicts with CA2000 (always dispose object before returning), which is also error-prone.
